common/djangoapps/third_party_auth/tests/test_models.py

```diff
@@ -17,39 +25,61 @@ def setUp(self):
-
-            # 7/21/22 : Disabling the exception on duplicate entity ID's because of existing data.
-            # with pytest.raises(IntegrityError):
-            bad_config = SAMLProviderConfig(entity_id=self.saml_provider_config.entity_id)
-            bad_config.save()
-        assert ctx.records[0].msg == f'Entity ID: {self.saml_provider_config.entity_id} already in use'
-
-    @override_settings(FEATURES={'ENABLE_UNICODE_USERNAME': False})
-    def test_clean_username_unicode_disabled(self):
-        """
-        Test the username cleaner function with unicode disabled
-        """
-        assert clean_username('ItJüstWòrks™') == 'ItJ_stW_rks'
+    @ddt.data(
+        ('ItJüstWòrks™', False, 'ItJ_stW_rks'),
+        ('ItJüstWòrks™', True, 'ItJüstWòrks'),
+        ('simple_username', False, 'simple_username'),
+        ('simple_username', True, 'simple_username'),
+        ('test@example.com', False, 'test_example_com'),
+        ('test@example.com', True, 'test@example.com'),
+    )
+    @ddt.unpack
+    def test_clean_username(self, input_username, unicode_enabled, expected_output):
+        """Test the username cleaner function with different unicode settings."""
+        with override_settings(FEATURES={'ENABLE_UNICODE_USERNAME': unicode_enabled}):
+            self.assertEqual(clean_username(input_username), expected_output)
+
 
-    @override_settings(FEATURES={'ENABLE_UNICODE_USERNAME': True})
-    def test_clean_username_unicode_enabled(self):
+@ddt.ddt
+class TestSAMLConfigurationSignals(TestCase):
+    """
+    Tests for SAML configuration signal handlers and their effect on provider configs.
+    """
+
+    def setUp(self):
         """
-        Test the username cleaner function with unicode enabled
+        Set up test data.
         """
-        assert clean_username('ItJüstWòrks™') == 'ItJüstWòrks'
+        self.site = Site.objects.get_current()
+        # Create initial SAML configuration
+        self.saml_config = SAMLConfiguration.objects.create(
+            site=self.site,
+            slug='test-config',
+            enabled=True,
+            entity_id='https://test.example.com',
+            org_info_str='{"en-US": {"url": "http://test.com", "displayname": "Test", "name": "test"}}'
+        )
+        # Create SAML provider that uses this configuration
+        self.provider_config = SAMLProviderConfig.objects.create(
+            site=self.site,
+            slug='test-provider',
+            enabled=True,
+            name='Test Provider',
+            entity_id='https://idp.test.com',
+            saml_configuration=self.saml_config
+        )
+        # Create some test SAML provider data
+        SAMLProviderData.objects.create(
+            entity_id='https://idp.test.com',
+            fetched_at='2023-01-01T00:00:00Z',
+            sso_url='https://idp.test.com/sso',
+            public_key='test-public-key'
+        )
+
+    @override_settings(ENABLE_SAML_CONFIG_SIGNAL_HANDLERS=True)
```

Comment:
1. Don't we want tests for enabled vs disabled?
2. We should have coverage for the custom attributes. See this past comment for help on implementing those assertions: https://github.com/openedx/edx-platform/pull/36954#discussion_r2229575556